dask/array/core.py

```diff
@@ -1850,6 +1851,19 @@ def normalize_chunks(chunks, shape=None):
     return tuple(tuple(int(x) if not math.isnan(x) else x for x in c) for c in chunks)
 
 
+def normalize_and_check_chunks(chunks, shape):
+    """Normalize chunks and check them for consistency."""
+    chunks = normalize_chunks(chunks, shape)
+    if len(chunks) != len(shape):
+        raise ValueError("Input array has %d dimensions but the supplied "
+                         "chunks has only %d dimensions" %
+                         (len(shape), len(chunks)))
+    if tuple(map(sum, chunks)) != shape:
+        raise ValueError("Chunks do not add up to shape. "
+                         "Got chunks=%s, shape=%s" % (chunks, shape))
```

Comment:
Should this check just be added to `normalize_chunks` instead?

Reply:
The checks currently aren't done in the `Array` constructor, which I assumed was an optimization to make creating an `Array` with internal dask.array functions less expensive. But if this doesn't meaningfully slow things down maybe it should be revisited.

Reply:
Timing for reference

```python
In [1]: chunks = ((1000,) * 100, (1000,) * 200)

In [2]: shape = (100000, 100000)

In [3]: %timeit tuple(map(sum, chunks)) != shape
The slowest run took 7.67 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 2.78 µs per loop
```

This seems fine to me.